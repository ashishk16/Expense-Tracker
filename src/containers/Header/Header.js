import React from 'react';
import {Link} from 'react-router-dom';
const Header = () => {
	return (
		<div className="header">
			<div className="header-background"/>
			<div className="nav-bar">
				<img className="col-1 logo" src={require('../../images/logo.png')} />
				<Link className="col-1" to="/"> Home</Link>
				<Link className="col-1" to={'/add'}> New Expense</Link>
				<Link className="col-1" to={'/add'}> All Expenses</Link>
				<Link className="col-1" to={'/add'}> About</Link>
			</div>
		</div>
	);
};

export default Header;
