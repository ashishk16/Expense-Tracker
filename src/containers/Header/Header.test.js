import Header from "./Header";
import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from "enzyme";
configure({adapter: new Adapter()});

describe('Header component', () => {
	it('should match snapshot',() => {
		const wrapper = shallow(<Header/>);
		expect(wrapper).toMatchSnapshot();
	});
});
