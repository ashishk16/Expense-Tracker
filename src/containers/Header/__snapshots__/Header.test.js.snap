// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="col-1 logo"
          src={Object {}}
        />,
        <Link
          className="col-1"
          replace={false}
          to="/"
        >
           Home
        </Link>,
        <Link
          className="col-1"
          replace={false}
          to="/all"
        >
           New Expense
        </Link>,
        <Link
          className="col-1"
          replace={false}
          to="/all"
        >
           All Expenses
        </Link>,
        <Link
          className="col-1"
          replace={false}
          to="/all"
        >
           About
        </Link>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "col-1 logo",
          "src": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": " Home",
          "className": "col-1",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": " Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": " New Expense",
          "className": "col-1",
          "replace": false,
          "to": "/all",
        },
        "ref": null,
        "rendered": " New Expense",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": " All Expenses",
          "className": "col-1",
          "replace": false,
          "to": "/all",
        },
        "ref": null,
        "rendered": " All Expenses",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": " About",
          "className": "col-1",
          "replace": false,
          "to": "/all",
        },
        "ref": null,
        "rendered": " About",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="col-1 logo"
            src={Object {}}
          />,
          <Link
            className="col-1"
            replace={false}
            to="/"
          >
             Home
          </Link>,
          <Link
            className="col-1"
            replace={false}
            to="/all"
          >
             New Expense
          </Link>,
          <Link
            className="col-1"
            replace={false}
            to="/all"
          >
             All Expenses
          </Link>,
          <Link
            className="col-1"
            replace={false}
            to="/all"
          >
             About
          </Link>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-1 logo",
            "src": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " Home",
            "className": "col-1",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": " Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " New Expense",
            "className": "col-1",
            "replace": false,
            "to": "/all",
          },
          "ref": null,
          "rendered": " New Expense",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " All Expenses",
            "className": "col-1",
            "replace": false,
            "to": "/all",
          },
          "ref": null,
          "rendered": " All Expenses",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": " About",
            "className": "col-1",
            "replace": false,
            "to": "/all",
          },
          "ref": null,
          "rendered": " About",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
